### PR TITLE
docs: initial github container registry integration guide

### PR DESCRIPTION
**Pull request by lukemarsden**
An initial guide to how to push to container images into GHCR from Dagger running in GitHub Actions.

### CODE REVIEW

**Comment by lukemarsden** [comment]
CI failure seems unrelated to this change?

File: docs/current_docs/integrations/snippets/ghcr-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Dagger Build & Push
+        uses: dagger/dagger-for-github@v5
+        with:
+          version: "0.11.0"
+          verb: call
+          args: build-and-push --registry=$DOCKER_REGISTRY --image-name=$DOCKER_IMAGE_NAME --username=$DOCKER_USERNAME --password=env:DOCKER_PASSWORD --build-context .
```
Comment (vikram-dagger) [inline docs/current_docs/integrations/snippets/ghcr-workflow.yml:23]:
```suggestion
          args: build-and-push --registry=$DOCKER_REGISTRY --image-name=$DOCKER_IMAGE_NAME --username=$DOCKER_USERNAME --password=env:DOCKER_PASSWORD --build-context=.
```

**Comment by vikram-dagger** [comment]
I've not reviewed in depth but a few questions off the top:
- Why don't we incorporate this into the existing GitHub Actions integrations page? This would avoid duplicating the GHA workflow. Having two workflow examples risks user confusion, as it's then not clear which version is canonical.  
- The file should be called `....go` instead of `...go.txt`. You could place it in a separate subdir of `snippets` and include the `dagger.gen.go` and `go.mod` files, and then you would not need to import `dagger` in the Go code. Example structure [here](https://github.com/dagger/dagger/blob/main/docs/current_docs/guides/snippets/create-app-ci-module/base/go/main.go).

**Comment by vikram-dagger** [comment]
> CI failure seems unrelated to this change?

CI error is `failed to get edge: inconsistent graph state` which does indeed seem unrelated.

**Comment by lukemarsden** [comment]
Yeah, happy to move it into the GHA page as a sub-section if you think that's best!

**Comment by vikram-dagger** [comment]
@lukemarsden OK to push changes directly to your PR?

**Comment by lukemarsden** [comment]
Sure please do!

lukemarsden.net
***@***.***
@lmarsden <https://twitter.com/lmarsden>

**Comment by lukemarsden** [comment]
Updated the module reference to point to `daggerverse/dagger-ghcr-demo` instead of `lukemarsden/dagger-ghcr-demo`

This LGTM

**Comment by vikram-dagger** [comment]
Ping @kpenfound for review

**Review by vikram-dagger — APPROVED** [review]

**Review by kpenfound — APPROVED** [review]
LGTM!

**Pull request merged**